lib/workers/repository/process/index.ts

```diff
@@ -12,13 +13,28 @@ import type { WriteUpdateResult } from './write';
 function getBaseBranchConfig(
   baseBranch: string,
   config: RenovateConfig
-): RenovateConfig {
+): Promise<RenovateConfig> {
   logger.debug(`baseBranch: ${baseBranch}`);
-  const baseBranchConfig = mergeChildConfig(config, { baseBranch });
-  if (config.baseBranches.length > 1) {
+
+  let baseBranchConfig: RenovateConfig = config;
+
+  if (
+    config.useBaseBranchConfig === 'replace' &&
+    baseBranch !== config.defaultBranch
+  ) {
+    baseBranchConfig = await platform.getJsonFile(
+      config.onboardingConfigFileName,
+      config.repository,
+      baseBranch
+    );
+  }
+
+  baseBranchConfig = mergeChildConfig(baseBranchConfig, { baseBranch });
```

Comment:
Sorry, it is not clear how to do that. Maybe you can give an example?

Reply:
@rarkins I agree merge is needed, because some base config fields are missing and it doesn't work on a real repo. I'm trying to fix.

Reply:
I'd like to ask for an advice from you. I was thinking how to merge the global config we have from the global worker and the branch config we get from `getJsonFile()` AND *remove* the master branch config we already parsed on the repo init stage. I see a possible solution: modify this function:

https://github.com/renovatebot/renovate/blob/51fa664ba42f75bab334d3de74d61bd612aac0ca/lib/workers/repository/init/merge.ts#L167

and add the second optional argument `branchName`. This function does exactly what I need - it merges the global config with the `renovate.json` file it reads from the default branch. Instead it will detect/read config file from the branch I'll specify. 

Then I'll need to call `mergeRenovateConfig()` and it will do the merge I need.

I'd like to confirm with you first before continue implementing this solution, because it may have side effects I can't see for now.

Reply:
Hi @rarkins could you please advice on this? I'd like to modify `mergerRenovateConfig()` since it does a lot of things with the config: if the config has `extends`, merges the extended parts, and others. I would proceed with this solution if you approve.

Reply:
I did my best try to merge the global config with the config read from branch. It complaints something about `fileMatch is not iterable` when I run it on the real repo. If you could review my latest changes and advice where I should improve the code, that would help me a lot. I'm a bit stuck with this one.

Reply:
Maybe you've some invalid config? can you share the configs? And a stacktrace.

Reply:
I saw that error when a regex manager had no `fileMatch` field

Reply:
After your help it started working on my real repo! Thanks a lot 👍